cli/src/commands/upload.ts

```diff
@@ -0,0 +1,183 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+
+import { BaseCommand } from '../cli/base-command';
+import { CrawledAsset } from '../cores/models/crawled-asset';
+import { CrawlService, UploadService } from '../services';
+import * as si from 'systeminformation';
+import FormData from 'form-data';
+import { UploadOptionsDto } from '../cores/dto/upload-options-dto';
+import { CrawlOptionsDto } from '../cores/dto/crawl-options-dto';
+
+export default class Upload extends BaseCommand {
+  private crawlService = new CrawlService();
+  private uploadService!: UploadService;
+  deviceId!: string;
+  uploadLength!: number;
+  dryRun = false;
+
+  public async run(paths: string[], options: UploadOptionsDto): Promise<void> {
+    await this.connect();
+
+    const uuid = await si.uuid();
+    this.deviceId = uuid.os || 'CLI';
+    this.uploadService = new UploadService(this.immichApi.apiConfiguration);
+
+    this.dryRun = options.dryRun;
+
+    const crawlOptions = new CrawlOptionsDto();
+    crawlOptions.pathsToCrawl = paths;
+    crawlOptions.recursive = options.recursive;
+    crawlOptions.excludePatterns = options.excludePatterns;
+
+    const crawledFiles: string[] = await this.crawlService.crawl(crawlOptions);
+
+    if (crawledFiles.length === 0) {
+      console.log('No assets found, exiting');
+      return;
+    }
+
+    const assetsToUpload = crawledFiles.map((path) => new CrawledAsset(path));
+
+    const cliProgress = require('cli-progress');
+    const byteSize = require('byte-size');
```

Comment:
Why are these required instead of imported using import statements above?

Reply:
Done